src/dotenv/main.py

```diff
@@ -35,7 +35,8 @@ def __init__(
         interpolate: bool = True,
         override: bool = True,
     ) -> None:
-        self.dotenv_path = dotenv_path  # type: Optional[Union[str, os.PathLike]]
+        # type: Optional[Union[str, os.PathLike]]
+        self.dotenv_path = dotenv_path
```

Comment:
Does the type hint really work when it's on a separate line? Is there an advantage to splitting this into several lines?